### PR TITLE
MAPPO ready for launch

### PR DESCRIPTION
**Pull request by prateekbhustali**
+ updated `self.world_state_size` in roadenv_wrapper.py
+ created MAPPO sweep config
+ updated default MAPPO config for sweep

created a test sweep and tested 1 run [here](https://wandb.ai/imp-act/jaxMARL-ToyExample-v2/runs/xyllvpm0?nw=nwusershaktimaan)

### CODE REVIEW

File: getting-started/config/hyper_parameter_tuning/toy_example_v2/mappo_rnn_road_env.yaml [hunk 1]
```diff
@@ -52,7 +52,8 @@ REWARD_STANDARDIZATION: True
 "WANDB_LOG_ALL_SEEDS": False # will log vmapped seeds separately
 
 # where to save the params (if None, will not save)
-"SAVE_PATH": null
+IF_SAVE: False
```
Comment (moratodpg) [inline getting-started/config/hyper_parameter_tuning/toy_example_v2/mappo_rnn_road_env.yaml]: Why do we include two variables (instead of the default option `if config.get("SAVE_PATH", None) is not None:` )? Now we have to modify two inputs in the config.

File: getting-started/config/hyper_parameter_tuning/toy_example_v2/mappo_sweep.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        map_name: ToyExample-v2
+      - encoding_type: none
+        include_extra_observations:
+          segment_lengths: true
+          volumes: true
+          capacity: false
+        map_name: ToyExample-v2
+
+program: getting-started/mappo_rnn_road_env.py
+project: jaxMARL-ToyExample-v2-test
```
Comment (moratodpg) [inline getting-started/config/hyper_parameter_tuning/toy_example_v2/mappo_sweep.yaml]: Shouldn't this be `jaxMARL-ToyExample-v2`?

**Comment by prateekbhustali** [comment]
good catch @moratodpg, I've fixed it, thanks!

File: getting-started/mappo_rnn_road_env.py [hunk 3]
```diff
@@ -758,16 +815,38 @@ def _greedy_env_step(step_state, unused):
...
+
         rng, _rng = jax.random.split(rng)
 
-        # Storing evaluation metrics on CPU to save GPU memory
+        # Metrics Manager
         num_evals = int(1 / config["TEST_INTERVAL"])
-        initial_eval_metrics = jnp.zeros((num_evals,), device="cpu")
+        metrics_manager = EvalMetricsManager(
+            logged_eval_metrics=get_greedy_metrics(_rng, train_states),
+            eval_returns=jnp.zeros((num_evals,), device="cpu"),
```
Comment (moratodpg) [inline getting-started/mappo_rnn_road_env.py:843]: Have you defined "cpu" intentionally here?

Comment (prateekbhustali) [reply getting-started/mappo_rnn_road_env.py:843]: yes! It was just to safeguard against possible OOMs when running large envs

**Review by moratodpg — COMMENTED** [review]
Very nice additions! Thanks, @omniscientoctopus ! It is working nicely on my machine. 

I think it would be good if @DanielHettegger also takes a look since there are some important modifications here (mainly related to eval metrics logging and model checkpointing).

One last quick minor comment:
QLearning methods also log the number of gradient steps `grad_steps`. (In case you would also like to include it). I checked it, but MAPPO's TrainState does not include `grad_steps`.

File: getting-started/mappo_rnn_road_env.py (hunk 3 above)
Comment (moratodpg) [reply getting-started/mappo_rnn_road_env.py:843]: Could this lead to a "communication GPU-CPU" overhead? (But I guess it will probably be tiny - since we are evaluating every once in a while)

**Review by DanielHettegger — APPROVED** [review]
Looks great, thanks a lot @omniscientoctopus!
